generator/internal/connection_generator.cc

```diff
@@ -192,10 +201,32 @@ Status ConnectionGenerator::GenerateHeader() {
   // close abstract interface Connection base class
   HeaderPrint("};\n");
 
-  HeaderPrint(  // clang-format off
-    "\nstd::shared_ptr<$connection_class_name$> Make$connection_class_name$(\n"
-    "    Options options = {});\n");
-  // clang-format on
+  HeaderPrint(R"""(
+/**
+ * A factory function to construct a `$connection_class_name$` object.
+ *
+ * The returned connection object should not be used directly; instead it
+ * should be given to a `$client_class_name$` instance, and methods should be
```

Comment:
"passed as an argument to the constructor of `$client_class_name$`, and that class used instead."

Reply:
Done.